enhancements/network/on-prem-dual-stack-vips.md

```diff
@@ -0,0 +1,421 @@
+but the cluster VIPs will be ipv4 (at this time ipv6 VIPs are not supported
+in dual stack clusters).
+
+### Goals
+
+Create both ipv4 and ipv6 VIPs on dual stack clusters so single stack
+applications have the connectivity they require.
+
+### Non-Goals
+
+Making the `kubernetes.default` Service dual stack is not supported and will
+not be addressed by this work. We are only adding dual stack access to the
+API via the VIPs.
+
+## Proposal
+
+### User Stories
+
+As a deployer of a dual stack OpenShift cluster, I want to run ipv6-only
+applications that need access to the API and ingress VIPs.
+
+### API Extensions
+
+New fields will have to be added to the
+[platform status section](https://github.com/openshift/api/blob/24043075985b3dc87190174241f3c4d0416b8ea2/config/v1/types_infrastructure.go#L465)
+of the infrastructure object to persist the specified additional VIPs.
+
+### Implementation Details/Notes/Constraints [optional]
+
+This should be a relatively simple feature to implement since most of the
+new code, manifests, and templates will be slight modifications of the
+existing loadbalancer ones. The main thing we need to determine is what the
+structure of the install-config and api changes will be. That will have an
+impact on the implementation of these new VIPs, but any of the designs
+proposed below should be doable. It's mostly a question of picking the one
+that will provide the best user experience and, secondarily, the simplest
+implementation.
+
+### Risks and Mitigations
+
+Minimal risk. This is just adding another VIP, something we already have in
+our deployments. The main concern would be logic errors arising out of
+the need to handle multiple VIPs, which (if they happen) will need to be
+addressed as bugs.
+
+## Design Details
+
+This feature will require changes in a few different components:
+
+Installer - New fields will need to be added to install-config for deployers
+to specify the additional VIPs.
+
+API - New fields will need to be added to the platform status section of the
+infrastructure object.
+
+machine-config-operator - The on-prem loadbalancer templates and manifests
+will need to be updated to support the additional VIPs.
+
+baremetal-runtimecfg - There is logic that depends on the ip version of the
+VIP. If there are multiple VIPs, this will have to be modified to account for
+that. In addition, the render code will need the new values wired in.
+
+### install-config
+
+Currently the VIPs are specified as follows:
+
+```yaml
+platform:
+  baremetal:
+    apiVIP: 192.168.111.5
+    ingressVIP: 192.168.111.4
+```
+
+After this change it will look more like this:
+
+```yaml
+platform:
+  baremetal:
+    apiVIP: 192.168.111.5
+    # Optional, and only for dual stack clusters
+    apiVIPSecondary: fd2e:6f44:5dd8:c956::5
+    ingressVIP: 192.168.111.4
+    # Optional, and only for dual stack clusters
+    ingressVIPSecondary: fd2e:6f44:5dd8:c956::4
```

Comment:
Can we call these new configs apiVIPv6 and ingressVIPv6 instead of calling them "Secondary" ? 

If it is necessary to mark one of them as primary & the other secondary, we should have another optional config like primaryApiVIP with a default value of apiVIP. Assuming the v6 VIPs as secondary seems arbitrary.